gpt4all-chat/src/database.cpp

```diff
@@ -2524,6 +2534,35 @@ void Database::retrieveFromDB(const QList<QString> &collections, const QString &
             results->append(tempResults.value(id));
 }
 
+bool Database::ftsIntegrityCheck()
+{
+    bool isConsistent = false;
+
+    QSqlQuery q(m_db);
+    if (!q.prepare("insert into chunks_fts(chunks_fts, rank) values('integrity-check', 1);")) {
+        qWarning() << "ERROR: Cannot prepare sql for fts integrity check" << q.lastError();
+        return false;
+    }
+
+    // Returns an error executing sql if it the integrity check fails
+    // See: https://www.sqlite.org/fts5.html#the_integrity_check_command
+    isConsistent = q.exec();
+
+    if (!isConsistent) {
+        if (!q.prepare("insert into chunks_fts(chunks_fts) values('rebuild');")) {
```

Comment:
If we don't use any bind values, why don't we exec in-place?
```suggestion
        if (!q.exec("insert into chunks_fts(chunks_fts) values('rebuild');")) {
```